### DISCUSSION TITLE
TS type including all relations

### DISCUSSION THREAD

**Discussion by unlocomqx** [prisma/prisma#8141] [Q&A] [answered]
Does prisma generate a type which includes all the model's relations?
I always have to write
```ts
let user: User & {profile: Profile, post: Post[], }
```
Is there a shorthand that is generated by prisma?

**Comment by ryands17** [comment, marked as answer]
@unlocomqx 👋 

You can do it as follows:

```ts
type UserWithPosts = Prisma.UserGetPayload<{
    include: { posts: true }
  }>
```

**Reply by unlocomqx** [reply]
Thank you, It's a neat approach!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
